### PR TITLE
Change the data structure to allow configurable master count

### PR DESCRIPTION
**Pull request by yrobla**
Send the node properties inside flexible maps, to allow different
number of masters in the future. Set a NUM_MASTERS var that defaults to 3,
with the possibility of changing to 1 in the future.

### CODE REVIEW

File: utils.sh [hunk 1]
```diff
@@ -94,58 +94,94 @@ function master_node_val() {
...
+      username=$(master_node_val ${master_idx} "driver_info.${driver_prefix}_username")
+      password=$(master_node_val ${master_idx} "driver_info.${driver_prefix}_password")
+      address=$(master_node_val ${master_idx} "driver_info.${driver_prefix}_address")
+
+      deploy_kernel=$(master_node_val ${master_idx} "driver_info.deploy_kernel")
+      deploy_ramdisk=$(master_node_val ${master_idx} "driver_info.deploy_ramdisk")
+
+      cat <<EOF
+        openshift-master-$master_idx:
+          ${port_prefix}${port_newline}${driver_prefix}_username: "${username}"
```
Comment (stbenjam) [inline utils.sh]: With an IPMI port, the indentation is off a little on username (causes an error):

```
      driver_infos:
        openshift-master-0:
          ipmi_port: "6230"
        ipmi_username: "admin"
          ipmi_password: "password"
          ipmi_address: "192.168.111.1"
          deploy_kernel:  "http://172.22.0.1/images/ironic-python-agent.kernel"
          deploy_ramdisk: "http://172.22.0.1/images/ironic-python-agent.initramfs"
```

Error:

```
level=fatal msg="failed to fetch Master Machines: failed to load asset \"Install Config\": failed to unmarshal: error converting YAML to JSON: yaml: line 49: did not find expected key"
```

File: utils.sh [hunk 2]
```diff
@@ -94,58 +94,94 @@ function master_node_val() {
...
+
+    cat <<EOF
+      driver_infos:
+EOF
+
+    for ((master_idx=0;master_idx<$1;master_idx++)); do
+      port=$(master_node_val ${master_idx} "driver_info.${driver_prefix}_port // \"\"")
+      if [ -n "$port" ]; then
+  	    port_prefix="${driver_prefix}_port: \"${port}\""
+        port_newline=$'\n        '
```
Comment (stbenjam) [inline utils.sh]: I think this needs an additional 2 spaces

**Comment by stbenjam** [comment]
Looks like there’s still conflicts with master

**Comment by hardys** [comment]
lgtm, will test locally and lets see if CI works (I updated the pull secret so this is a test to see if CI is now working, lets not block on it if there are other issues that cause the job to fail).

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/580/

**Comment by hardys** [comment]
Hmm the CI error seems real, does this depend on a kni-installer patch?

```
Apr 18 08:49:16 |  level=debug msg="2019/04/18 08:49:16 [DEBUG] New state was assigned lineage \"9c0ca805-4995-f86b-c3bb-15ecea6175bd\""
Apr 18 08:49:19 |  level=error
Apr 18 08:49:19 |  level=error msg="Error: variable driver_info_2 should be type map, got string"
Apr 18 08:49:19 |  level=error
Apr 18 08:49:19 |  level=error
Apr 18 08:49:19 |  level=error
Apr 18 08:49:19 |  level=error msg="Error: variable root_device_2 should be type map, got string"
...
```

**Comment by hardys** [comment]
We could add https://patch-diff.githubusercontent.com/raw/openshift-metalkube/kni-installer/pull/47.patch to 03_ocp_repo_sync.sh perhaps to see if we get a clean CI run?

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/581/

**Comment by derekhiggins** [comment]
Build FAILURE, see build http://10.8.144.11:8080/job/dev-tools/582/

**Comment by dantrainor** [comment]
Do we have a plan for using NUM_MASTERS=1 in the near future?

**Comment by yrobla** [comment]
There is, although by default will be 3. But there is still a problem with mdns to allow 1.

**Pull request merged**